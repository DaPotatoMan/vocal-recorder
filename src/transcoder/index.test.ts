import OGGLargeAudioURL from '~/tests/assets/audio-sample-large.ogg?url'
import M4AAudioURL from '~/tests/assets/audio-sample.m4a?url'
import MP3AudioURL from '~/tests/assets/audio-sample.mp3?url'
import WAVAudioURL from '~/tests/assets/audio-sample.wav?url'
import WEBMAudioURL from '~/tests/assets/audio-sample.webm?url'

import { Transcoder } from '..'

describe('class: Transcoder', async () => {
  async function transcode(url: string) {
    const blob = await fetch(url).then(i => i.blob())

    const onProgressFn = vi.fn((value: number) => console.log(`Progress: ${(value * 100).toFixed(2)}%`))
    const result = await Transcoder.toMP3(blob, {
      onProgress: onProgressFn
    })

    expect(onProgressFn).toHaveBeenCalled()
    expect(result).toMatchSnapshot()
  }

  it('mp3 -> mp3', async () => transcode(MP3AudioURL))
  it('wav -> mp3', async () => transcode(WAVAudioURL))
  it('webm -> mp3', async () => transcode(WEBMAudioURL))
  it.skip('m4a -> mp3', async () => transcode(M4AAudioURL))
  it('ogg (large) -> mp3', async () => transcode(OGGLargeAudioURL), 25_000)
})
