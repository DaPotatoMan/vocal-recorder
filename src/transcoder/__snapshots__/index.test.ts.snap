// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`class: Transcoder > m4a -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 80,
    "duration": 23.92526,
  },
}
`;

exports[`class: Transcoder > mp3 -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 128,
    "duration": 23.902,
  },
}
`;

exports[`class: Transcoder > ogg (large) -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 112,
    "duration": 1690.867667,
  },
}
`;

exports[`class: Transcoder > wav -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 320,
    "duration": 23.902041,
  },
}
`;

exports[`class: Transcoder > webm (large) -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 112,
    "duration": 1690.867667,
  },
}
`;

exports[`class: Transcoder > webm -> mp3 1`] = `
{
  "blob": Blob {},
  "metadata": {
    "bitRate": 80,
    "duration": 23.91,
  },
}
`;
